### PR TITLE
Refactor $WEB_CONCURRENCY handling

### PR DESCRIPTION
**Pull request by dzuelke**
Changes the various functions around so they no longer have side effects.

### CODE REVIEW

File: profile/WEB_CONCURRENCY.sh [hunk 1]
```diff
@@ -3,16 +3,25 @@
...
   local web_memory=$2
-  local concurrency
 
-  concurrency=${WEB_CONCURRENCY-$(($available/$web_memory))}
+  echo $(($available/$web_memory))
+}
+
+validate_concurrency() {
+  local concurrency=$1
+  local oob=0
```
Comment (Malax) [inline profile/WEB_CONCURRENCY.sh:12]: What does `oob` mean?

**Review by Malax — APPROVED** [review]

File: profile/WEB_CONCURRENCY.sh (hunk 1 above)
Comment (dzuelke) [reply profile/WEB_CONCURRENCY.sh:12]: "out of bounds"; it's the return value (as opposed to function result, which is what gets `echo`'d so you can capture it with `$()`) that lets calling code determine whether the calculated (or given) was too high or too low

Comment (dzuelke) [reply profile/WEB_CONCURRENCY.sh:12]: That's why the "`#FIXME`" comment is there - we could also change the warning function to `warn_high_concurrency`, removing the logic that checks for `> 200` (which right now is still duplicated)

Comment (Malax) [reply profile/WEB_CONCURRENCY.sh:12]: Got it now, thanks. Maybe worth adding a code comment about it? We could also eliminate the name `oob` completely as it's really just the return value. `local return_value` maybe? I was looking for a deeper meaning beyond "1 is lower bound exceeded and 2 is upper bound exceeded".

**Review by Malax — APPROVED** [review]

**Review by Malax — APPROVED** [review]

**Pull request merged**